data/README.md

```diff
@@ -22,4 +22,41 @@ Registers a `listener` function called everytime the state is updated.
 
 #### `store.dispatch( action: object )`
 
-The dispatch function should be called to trigger the registered reducers function and update the state. An `action` object should be passed to this action. This action is passed to the registered reducers in addition to the previous state.
\ No newline at end of file
+The dispatch function should be called to trigger the registered reducers function and update the state. An `action` object should be passed to this action. This action is passed to the registered reducers in addition to the previous state.
+
+
+### `wp.data.registerSelectors( key: string, selectors: object )`
+
+If your module or plugin needs to expose its state to other modules and plugins, you'll have to register state seclectors.
+
+A selector is function that takes the current state value as a first argument and extra arguments if needed and returns any data extracted from the state.
+
+#### Example:
+
+Let's say the state of our plugin (registered with the key `myPlugin`) has the following shape: `{ title: 'My post title' }`. We can register a `getTitle` selector to make this state value available like so:
+
+```js
+wp.data.registerSelectors( 'myPlugin', { getTitle: ( state ) => state.title } );
+```
+
+#### `wp.data.select( key: string, selectorName: string, ...args )`
+
+This function allows calling any registered selector. Given a module's key, a selector Name and extra arguments passed to the selector, this function calls the selector passing it the current state and the extra args provided.
+
+#### Example:
+
+```js
+wp.data.select( 'myPlugin', 'getTitle' ); // Returns "My post title"
+```
+
+#### `wp.data.query`
+
+If you use React or WordPress Element, a Higher Order Component is made available to inject data to your components like so:
+
+```js
+wp.data.query( select => {
+	return {
+		title: select( 'myPlugin', 'getTitle' ),
+	};
+} )( ( { title } ) => <div>{ title }</div> );
```

Comment:
I would suggest making it more focused on the `query` HOC by moving `( { title } ) => <div>{ title }</div>` to its own variable like this:
```js
const Component = ( { title } ) => <div>{ title }</div>;

wp.data.query( select => {
 	return {
 		title: select( 'myPlugin', 'getTitle' ),
 	};
 } )( Component  );
```